src/native/corehost/hostmisc/utils.cpp

```diff
@@ -358,12 +408,18 @@ bool get_dotnet_root_from_env(pal::string_t* dotnet_root_env_var_name, pal::stri
         return true;
 
 #if defined(WIN32)
-    if (pal::is_running_in_wow64())
+    if (is_running_in_wow64())
     {
         *dotnet_root_env_var_name = _X("DOTNET_ROOT(x86)");
         if (get_file_path_from_env(dotnet_root_env_var_name->c_str(), recv))
             return true;
     }
+    else if (is_emulating_x64())
+    {
+        *dotnet_root_env_var_name = _X("DOTNET_ROOT_x64");
+        if (get_file_path_from_env(dotnet_root_env_var_name->c_str(), recv))
+            return true;
+    }
```

Comment:
Why do we need this branch? The function already tries `DOTNET_ROOT_X64` above (since `get_arch` will always be `x64` when `is_emulating_x64` returns true).